src/tree/map.rs

```diff
@@ -1132,21 +1113,18 @@ fn split<K, V>(node: &mut Box<TreeNode<K, V>>) {
 // at input current key and returns search_key cmp cur_key
 // (i.e. search_key.cmp(&cur_key))
 fn tree_find_with<K, V, F>(
-    node: &Option<Box<TreeNode<K, V>>>,
+    mut node: &Option<Box<TreeNode<K, V>>>,
     mut f: F,
 ) -> Option<&V> where
     F: FnMut(&K) -> Ordering,
 {
-    let mut current = node;
     loop {
```

Comment:
It would be about the same:
```rust
while let Some(ref r) = *node {                                                       
    node = match f(&r.key) {                                                          
        Less => &r.left,                                                              
        Greater => &r.right,                                                          
        Equal => return Some(&r.value),                                               
    };                                                                                
}                                                                                     

None